### ISSUE TITLE
More descriptive error when missing __init__.py

### ISSUE THREAD

**Issue by Adrian-at-CrimsonAuzre** [PyCQA/pylint-django#352]
When an `__init__.py` file is missing in a module, pylint will crash.

The command I used was:
```
pylint --rcfile=pylintrc backend/core/serializers/some_serializer.py
```
which caused pylint to crash with the error:
```
django.core.exceptions.ImproperlyConfigured: Requested setting LOGGING_CONFIG, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
```
In this case, `__init__.py` was missing from the `backend/core/serializers/` folder. It took quite a bit of banging my head against a wall to realize my boneheaded mistake, but the misleading error message doesn't exactly help. Would it be possible to give a more descriptive error?

I can provide more detailed logs or an example project to reproduce this if you'd like.

**Comment by Adrian-at-CrimsonAuzre** [comment]
Funnily enough, I can't recreate it in our local Docker environment but It consistently happens with our Github action. Which is part of the reason I was banging my head against a wall for so long.

I've recreated a super basic example in [this repo](https://github.com/Adrian-at-CrimsonAuzre/testproject) and [here's an example](https://github.com/Adrian-at-CrimsonAuzre/testproject/runs/5190307722?check_suite_focus=true) of a failing test. The runner is Ubuntu 20.04, and as soon as `utils/__init__.py` is created pylint checks it without issue.

**Comment by Adrian-at-CrimsonAuzre** [comment]
Also, this issue doesn't present itself if pylint is run on the entire repo, only when it's checking an individual file within a directory missing `__init__.py`. `pylint --rcfile=pylintrc **/*.py` processes `utils/some_util.py` with no issue.
